projects/wvr-elements/src/lib/core/manifest/manifest.reducers.spec.ts

```diff
@@ -0,0 +1,330 @@
+      expect(JSON.stringify(fromManifestReducers.adapter.upsertOne(manifest, state).entities[0]) ===
+      JSON.stringify(fromManifestReducers.reducer(state, upsertManifestObj).entities[0]) )
+      .toBe(true);
+    });
+
+    // add many manifests
+    const manifests = [manifest, manifest];
+    const addManifestsObj = {
+      "manifests":[
+        {"name":"Directory App","baseUrl":"https://api-dev.library.tamu.edu/directory-service","entries":[{"name":"All Sorted","path":"/ldap/all-sorted","methods":["GET"]},{"name":"Departments","path":"/ldap/departments","methods":["GET"]}]},
+        {"name":"Directory App","baseUrl":"https://api-dev.library.tamu.edu/directory-service","entries":[{"name":"All Sorted","path":"/ldap/all-sorted","methods":["GET"]},{"name":"Departments","path":"/ldap/departments","methods":["GET"]}]}
+      ],
+      "type":"[Manifest] Add Manifests"
+    };
+
+    it("should have manifest action type as '[Manifest] Add Manifests'", () => {
+      expect(JSON.stringify(addManifestsObj.type) === JSON.stringify('[Manifest] Add Manifests'))
+      .toBe(true);
+    });
+
+    it('should add manifests', () => {
+      expect(JSON.stringify(fromManifestReducers.adapter.addMany(manifests, state).entities[0]) ===
+      JSON.stringify(fromManifestReducers.reducer(state, addManifestsObj).entities[0]) )
+      .toBe(true);
+    });
+
+    // upsert manifests
+    const upsertManifestsObj = {
+      "manifests":[
+        {"name":"Directory App","baseUrl":"https://api-dev.library.tamu.edu/directory-service","entries":[{"name":"All Sorted","path":"/ldap/all-sorted","methods":["GET"]},{"name":"Departments","path":"/ldap/departments","methods":["GET"]}]},
+        {"name":"Directory App","baseUrl":"https://api-dev.library.tamu.edu/directory-service","entries":[{"name":"All Sorted","path":"/ldap/all-sorted","methods":["GET"]},{"name":"Departments","path":"/ldap/departments","methods":["GET"]}]}
+      ],
+      "type":"[Manifest] Upsert Manifests"
+    };
+    it("should have manifest action type as '[Manifest] Upsert Manifests'", () => {
+      const upsertManifestsAction = fromManifestActions.upsertManifests({manifests});
+      expect(JSON.stringify(upsertManifestsAction.type) === JSON.stringify('[Manifest] Upsert Manifests'))
+      .toBe(true);
+    });
+
+    it('should upsert manifests', () => {
+      expect(JSON.stringify(fromManifestReducers.adapter.upsertMany(manifests, state).entities[0]) ===
+      JSON.stringify(fromManifestReducers.reducer(state, upsertManifestsObj).entities[0]) )
+      .toBe(true);
+    });
+
+    const update: Update<Manifest> = {
+      id: JSON.stringify( fromManifestReducers.adapter.selectId ),
+      changes: {name: JSON.stringify( fromManifestReducers.selectManifestByName(manifest))}
+    };
+
+    // update one manifest
+    const updateOneAdpater = fromManifestReducers.adapter.updateOne(
+      {id: JSON.stringify( fromManifestReducers.adapter.selectId ),changes: {name: manifest.name}}, state);
```

Comment:
Confirm that this `JSON.stringify` is needed

Reply:
needed.